mm2src/mm2_main/src/lp_swap/recreate_swap_data.rs

```diff
@@ -382,6 +404,14 @@ async fn recreate_taker_swap(ctx: MmArc, maker_swap: MakerSavedSwap) -> Recreate
             coin: maker_coin_ticker,
         })?;
 
+    #[cfg(all(not(target_arch = "wasm32"), feature = "new-db-arch"))]
+    {
+        taker_swap.maker_address = negotiated_event
+            .maker_coin_htlc_pubkey
+            .and_then(|pubkey| maker_coin.address_from_pubkey(&pubkey).ok())
+            .unwrap_or("Couldn't get the maker coin address. Please set it manually.".to_string());
```

Comment:
Shouldn't this be `expect` instead?

Reply:
so this goes as follows: if we aren't able to get the address from pubkey for any reason. we will print this message in the resulting json as a placeholder, instead of failing.

who ever is importing the swap should manually edit the `[m/t]aker_address` to their own correct address to get a valid `[M/T]akerSavedSwap`.

this is only temporary, i plan in the future to include only the htlc pubkey (which is nonfallible), and let the importer of the recreated-swap convert such pubkey to an address. this avoids the cases where we can't generate the address from a simple pubkey (like zcoin) and the cases where we don't currently have this coin enabled to convert its pubkey to an address.

Reply:
Thanks, let's keep this "unresolved" to keep it visible as a future ref.